docs/build-hibench.md

```diff
@@ -48,3 +53,19 @@ Supported modules includes: micro, ml(machine learning), sql, websearch, graph,
 For Spark 2.0 and Spark 2.1, we add the benchmark support for Structured Streaming. This is a new module which cannot be compiled in Spark 1.6. And it won't get compiled by default even if you specify the spark version as 2.0 or 2.1. You must explicitly specify it like this:
 
     mvn -Psparkbench -Dmodules -PstructuredStreaming clean package 
+
+### Build using JDK 1.8
+If you are interested in building using Java 11 specify scala, spark and hadoop version as below
+
+    mvn clean package -Dhadoop=3.2 -Dspark=2.4 -Dscala=2.11
+
+Supported frameworks only: hadoopbench, sparkbench, (Not yet tested flinkbench, stormbench, gearpumpbench)
+Supported modules includes: micro, ml(machine learning), websearch and graph (not tested streaming and structuredStreaming) (Does not support sql)
+
+### Build using JDK 1.11
+If you are interested in building using Java 11 indicate that streaming benchmarks won't be compiled also, specify scala, spark and hadoop version as below
+
+    mvn clean package -Psparkbench -Phadoopbench -Dhadoop=3.2 -Dspark=2.4 -Dscala=2.12 -Dexclude-streaming
+
+Supported frameworks only: hadoopbench, sparkbench (Does not support flinkbench, stormbench, gearpumpbench)
+Supported modules includes: micro, ml(machine learning), websearch and graph (does not support streaming, structuredStreaming and sql)
```

Comment:
- What problem did SQL module meet? It's an essential part of Spark, leaving it alone makes not much sense : )
- Structured Streaming is a part of SQL, so making SQL work can also benefit SS

Reply:
**Q:** What problem did SQL module meet?
**A:** For newer versions of Spark HiveContext is deprecated, I can point out in de documentation that if `-Dspark=2.4` or further versions required then SQLBench won't work.
_(Again here it is necessary an update/port of ScalaSparkSQLBench.scala code)_
```
HiveContext is deprecated

In Spark 2, HiveContext is deprecated. Replace all usage with an instantiation of the singletonSparkSession:
val spark: SparkSession = SparkSession.builder
.config(conf)
.enableHiveSupport()
.getOrCreate()
Most functionality of HiveContext is now available directly on the SparkSession instance. Note that, if you need them, SparkContext and SQLContext are now properties of SparkSession:
val sc = spark.sparkContext
val sqlContext = spark.sqlContext
```
[Info Source](https://medium.com/@engnews/migrating-our-scala-codebase-to-spark-2-d8b68759800a)

Reply:
Thanks I got it.
I think supporting Spark 2.4 without SQL module is quite weird. I can think of 2 ways:
- Drop Spark 1.6 support and modify the `ScalaSparkSQLBench.scala` to use `SparkSession`, which is introduced in Spark 2.0
- Create another seperate `ScalaSparkSQLBench`, deciding which class to use by Spark version

I like the first one better, newer HiBench version should drop some old codebase. cc @carsonwang